### DISCUSSION TITLE
Database calculated fields not working correctly

### DISCUSSION THREAD

**Discussion by benrshep** [Budibase/budibase#12959] [Help]
Hi, 

I am trying to use calculated fields in a table and they are not working as I expected. 

Here are my tables:
**Products Table**
<img width="715" alt="Screenshot 2024-02-05 at 3 14 02 PM" src="https://github.com/Budibase/budibase/assets/263140/a944bd7e-9ef0-4c81-86a9-f0e88319385f">
**Options Table**
<img width="904" alt="Screenshot 2024-02-05 at 3 14 11 PM" src="https://github.com/Budibase/budibase/assets/263140/6243a154-0e2a-4ff7-b415-0de11930bb32">
**Formula settings**
<img width="531" alt="Screenshot 2024-02-05 at 3 14 27 PM" src="https://github.com/Budibase/budibase/assets/263140/9d0992d4-ebc3-4a67-8198-b81273f25bed">
**Formula**
<img width="1005" alt="Screenshot 2024-02-05 at 3 14 33 PM" src="https://github.com/Budibase/budibase/assets/263140/c60c4d2d-9dbd-4a78-9bd7-2e47523657b8">

As you can see from the **Options Table** the returned values are not correct. It is returning "Classroom Desk: Adjustable" for all of the rows. Whereas it should be different for each row. 

Oddly, it does calculate correctly if I create a new row, or if I edit an existing row. But if I refresh the page, they all update to the same value.

**Comment by ConorWebb96** [comment]
Hey @benrshep,

With this being a simple formula I'd recommend using handleBars. However, regardless of what you use it should make any difference to the results outputted. 

I was unable to replicate your issue. I'm on the latest version of Budibase, so there may be a chance that this was indirectly fixed. I'd definitely also recommend remaking the formula using a static formula type. As if you are wanting to filter on this column later you will need it to be a static formula.

<img width="1510" alt="Screenshot 2024-02-08 at 08 47 45" src="https://github.com/Budibase/budibase/assets/126772285/99309319-f6e3-4a77-895c-9e9337e55818">
<img width="1512" alt="Screenshot 2024-02-08 at 08 47 51" src="https://github.com/Budibase/budibase/assets/126772285/2fa2a5aa-79f6-4818-b685-d173dfe77817">
<img width="1512" alt="Screenshot 2024-02-08 at 08 47 37" src="https://github.com/Budibase/budibase/assets/126772285/3fbe2627-1e6e-4ad9-b4ba-f0214b002bca">

Even with switching to JS it still seems to work as intended. Could you be on an older version of Budibase?

<img width="1398" alt="Screenshot 2024-02-08 at 08 50 17" src="https://github.com/Budibase/budibase/assets/126772285/946a9af9-ab3b-44e3-8ebf-4a303cd22643">
<img width="1389" alt="Screenshot 2024-02-08 at 08 50 25" src="https://github.com/Budibase/budibase/assets/126772285/d9d8249f-05d7-4bea-9c68-267b20a92803">
